source/backend/api/Services/DispositionFileService.cs

```diff
@@ -52,12 +55,14 @@ public class DispositionFileService : IDispositionFileService
             _propertyService = propertyService;
             _lookupRepository = lookupRepository;
             _checklistRepository = checklistRepository;
+            _userRepository = userRepository;
         }
 
         public PimsDispositionFile Add(PimsDispositionFile dispositionFile, IEnumerable<UserOverrideCode> userOverrides)
         {
             _logger.LogInformation("Creating Disposition File {dispositionFile}", dispositionFile);
             _user.ThrowIfNotAuthorized(Permissions.DispositionAdd);
+            dispositionFile.ThrowMissingContractorInTeam(_user, _userRepository);
```

Comment:
is this function logically equivalent to: _user.ThrowInvalidAccessToDispositionFile(_userRepository, _dispositionFileRepository, id);
?
If so I'd rather not have two functions that provide the same functionality. Especially given that this function seems to require that the disposition team be provided with the file which is not a guarantee.